independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverInfo.java

```diff
@@ -123,12 +126,27 @@ int getPriority() {
 
     MethodParameterInfo initEventParam(MethodInfo observerMethod) {
         List<MethodParameterInfo> eventParams = new ArrayList<>();
+
         for (AnnotationInstance annotation : declaringBean.getDeployment().getAnnotations(observerMethod)) {
             if (Kind.METHOD_PARAMETER == annotation.target().kind()
                     && (annotation.name().equals(DotNames.OBSERVES) || annotation.name().equals(DotNames.OBSERVES_ASYNC))) {
                 eventParams.add(annotation.target().asMethodParameter());
+
+                /**
+                 * Log a warning for usage of transactional observers other than IN_PROGRESS since they are not implemented yet:
+                 * https://github.com/quarkusio/quarkus/issues/2224.
+                 */
+                final AnnotationValue during = annotation.value("during");
+                if (isTransactionalObserver(during)) {
```

Comment:
What about `if (isTransactionalObserver(during) && !isDuringTransactionInProgress(during))`?

Reply:
I'll push the change shortly.